packages/aws-cdk-lib/aws-cloudfront/lib/private/cache-behavior.ts

```diff
@@ -27,6 +27,10 @@ export class CacheBehavior {
   constructor(originId: string, private readonly props: CacheBehaviorProps) {
     this.originId = originId;
 
+    if (props.enableGrpc && props.allowedMethods !== AllowedMethods.ALLOW_ALL) {
+      throw new Error('\'allowedMethods\' can only be AllowedMethods.ALLOW_ALL if \'enableGrpc\' is true');
```

Comment:
Otherwise CFn will fail:

```
Resource handler returned message: "Invalid request provided: AWS::CloudFront::Distribution: The parameter gRPC only supports the POST method. (Service: CloudFront, Status Code: 400, Request ID: ...
```

Reply:
I don't write like the following:

```ts
throw new Error(`'allowedMethods' can only be AllowedMethods.ALLOW_ALL if 'enableGrpc' is true, got: ${props.allowedMethods?.methods.join(', ')}`);
```

Because the `AllowedMethods.methods` is an array type and the message will not be intuitive.

```ts
export class AllowedMethods {
  /** HEAD and GET */
  public static readonly ALLOW_GET_HEAD = new AllowedMethods(['GET', 'HEAD']);
  /** HEAD, GET, and OPTIONS */
  public static readonly ALLOW_GET_HEAD_OPTIONS = new AllowedMethods(['GET', 'HEAD', 'OPTIONS']);
  /** All supported HTTP methods */
  public static readonly ALLOW_ALL = new AllowedMethods(['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE']);

  /** HTTP methods supported */
  public readonly methods: string[];

  private constructor(methods: string[]) { this.methods = methods; }
}
```

Otherwise, if the user specifies `AllowedMethods.ALLOW_GET_HEAD_OPTIONS`, the following message will be displayed:

```
'allowedMethods' can only be AllowedMethods.ALLOW_ALL if 'enableGrpc' is true, got: 'GET', 'HEAD', 'OPTIONS'
```